### PR TITLE
(WIP) Package Commerce SDK for integration tests

### PR DESCRIPTION
**Pull request by dbarbato-sf**

### CODE REVIEW

**Comment by dbarbato-sf** [comment]
- [ ] Confirm nyc is creating coverage for the sdk.
- [ ] Add a call to run the integration tests through top level package.json

File: integration-tests/test_integration/client-integration.test.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
+/*
```
Comment (monkey-jeff) [inline integration-tests/test_integration/client-integration.test.ts:1]: [NIT] I don't like seeing both kebab case and snake case in the same path.

File: integration-tests/test_integration/client-integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+ * Copyright (c) 2019, salesforce.com, inc.
```
Comment (monkey-jeff) [inline integration-tests/test_integration/client-integration.test.ts:2]: Shouldn't this be 2020?

File: integration-tests/test_integration/client-integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+import { Checkout} from "commerce-sdk";
+import chai, { expect } from "chai";
+import chaiAsPromised from "chai-as-promised";
+import { assert, IsExact } from "conditional-type-checks";
+import { commonParameterPositions } from "@commerce-apps/core";
+
+const BASE_URI =
```
Comment (monkey-jeff) [inline integration-tests/test_integration/client-integration.test.ts:14]: We can remove the setting of the base URI for this test.  (And I would love to not set custom baseUri for most default integration tests calls)

File: integration-tests/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+  "name": "commerce-sdk-integration-test",
+  "version": "1.4.1-beta.0",
+  "description": "Salesforce Commerce Cloud SDK Integration Test",
+  "homepage": "https://developer.commercecloud.com/s/",
+  "scripts": {
```
Comment (monkey-jeff) [inline integration-tests/package.json:6]: Going to assume these will be cleaned up.

File: integration-tests/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    "test:integration:redis": "fetch:internal && npm run redis --prefix ../packages/core && npm run test; npm run redis:shutdown --prefix ../packages/core",
+    "test:integration:redis:debug": "fetch:internal && npm run redis --prefix ../packages/core && npm run test; npm run redis:shutdown:debug --prefix ../packages/core"
+  },
+  "dependencies": {
+    "@commerce-apps/core": "file:./commerce-apps-core-1.4.1-beta.0.tgz",
+    "commerce-sdk": "file:./commerce-sdk-1.4.1-beta.0.tgz"
+  },
+  "nyc": {
+    "extends": "@istanbuljs/nyc-config-typescript",
+    "include": [
```
Comment (monkey-jeff) [inline integration-tests/package.json:19]: Interesting approach...I kind of like it!

File: integration-tests/package.json [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  "homepage": "https://developer.commercecloud.com/s/",
+  "scripts": {
+    "test": "nyc --check-coverage=false mocha \"test_integration/*.test.ts\" --retries 2 --timeout 5000 --reporter=xunit --reporter-options output=../reports/integration.xml",
+    "fetch:internal": "npm pack ../packages/core && npm pack ../packages/generator && npm install ",
+    "test:integration": "npm run fetch:internal &&  npm run test",
+    "test:integration:redis": "fetch:internal && npm run redis --prefix ../packages/core && npm run test; npm run redis:shutdown --prefix ../packages/core",
+    "test:integration:redis:debug": "fetch:internal && npm run redis --prefix ../packages/core && npm run test; npm run redis:shutdown:debug --prefix ../packages/core"
+  },
+  "dependencies": {
+    "@commerce-apps/core": "file:./commerce-apps-core-1.4.1-beta.0.tgz",
```
Comment (monkey-jeff) [inline integration-tests/package.json:14]: How do these files get here?

File: integration-tests/test_integration/client-integration.test.ts (hunk 1 above)
Comment (dbarbato-sf) [reply integration-tests/test_integration/client-integration.test.ts:1]: Yep, agreed, will clean this up before putting the PR in initial review.

File: integration-tests/test_integration/client-integration.test.ts (hunk 2 above)
Comment (dbarbato-sf) [reply integration-tests/test_integration/client-integration.test.ts:2]: Copy paste error, does it need to be cleaned up across the board?

File: integration-tests/package.json (hunk 5 above)
Comment (dbarbato-sf) [reply integration-tests/package.json:19]: I still need to prove it works in this case.

File: integration-tests/package.json (hunk 6 above)
Comment (dbarbato-sf) [reply integration-tests/package.json:14]: npm pack, done in the fetch:internal script, creates the them and drops them in the current working directory, in this case the integration-tests folder.

File: integration-tests/test_integration/client-integration.test.ts (hunk 2 above)
Comment (monkey-jeff) [reply integration-tests/test_integration/client-integration.test.ts:2]: It's a "file gets touched, update it" kind of situation.  Sometimes it gets missed.

File: integration-tests/test_integration/client-integration.test.ts (hunk 1 above)
Comment (unandyala) [reply integration-tests/test_integration/client-integration.test.ts:1]: We earlier decided on lower camelCase for all the file names. And we named all the source files in one PR.

File: integrationTests/package-lock.json [hunk 7]
```diff
@@ -0,0 +1,9560 @@
+{
```
Comment (dbarbato-sf) [inline integrationTests/package-lock.json:1]: @monkey-jeff @bredmond-sf Should this get added to .gitignore? The others don't seem to be, but they're also generated...

Comment (kalagarraj) [reply integrationTests/package-lock.json:1]: Have you tried npm pack with package-lock already present? It causes integrity errors for the new pack files during npm install step

**Pull request closed without merging**